CHANGELOG.md

```diff
@@ -1,3 +1,27 @@
+## 2.12.0-rc.0
+
+* [FEATURE] Add PromQL query logging. #5794
+* [FEATURE] Enable and provide binaries for `mips64` / `mips64le` architectures. #5792
+* [ENHANCEMENT] Mark deleted rule's series as stale on next evaluation. #5759
+* [ENHANCEMENT] Check for duplicate label names in remote read. #5829
+* [ENHANCEMENT] Add more details about retention to storage docs. #5842
+* [ENHANCEMENT] Improve responsiveness of targets web UI and API endpoint. #5740
+* [ENHANCEMENT] Improve remote write desired shards calculation. #5763
+* [ENHANCEMENT] Flush TSDB pages more precisely. tsdb#660
+* [ENHANCEMENT] Add `prometheus_tsdb_retention_limit_bytes` metric. tsdb#667
+* [ENHANCEMENT] Add logging during TSDB WAL replay on startup. tsdb#662
+* [ENHANCEMENT] Improve TSDB memory usage. tsdb#653, tsdb#643, tsdb#654, tsdb#642, tsdb#627
+* [BUGFIX] Fix JavaScript error when showing warning about out-of-sync server time. #5833
+* [BUGFIX] Fix example command lines for starting `remote_write_adapter` in docs. #5830
+* [BUGFIX] Correct function name in remote write queue log message. #5852
```

Comment:
This is too small to mention I'd say, plus a few of the others. It doesn't affect the server's semantics.